gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagerUtils.java

```diff
@@ -307,6 +307,12 @@ static int getDagQueueId(long flowExecutionId, int numThreads) {
     return (int) (flowExecutionId % numThreads);
   }
 
+  static boolean isDagFromAdhocFlow(Dag<JobExecutionPlan> dag) {
+    // Every dag should have at least one node, and every dag will clone the job configurations which include the schedule
+    // defaults to false (so metrics are still tracked) if the dag property is not configured due to old dags
+    return ConfigUtils.getBoolean(dag.getStartNodes().get(0).getValue().getJobSpec().getConfig(), ConfigurationKeys.GOBBLIN_FLOW_ISADHOC,false);
```

Comment:
if `DagMgrUtils` doesn't yet have an abstraction for getting a DAG's job config, we might consider one:
```
.getStartNodes().get(0).getValue().getJobSpec().getConfig()
```